client/lib/config/index.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Creates config API with data applied.
+ *
+ * @param {Object} data Config data
+ * @return {Function} Config API
+ */
+const createConfig = data => {
+	/**
+	 * Retrieves property from config by `key`.
+	 * Throws an error if the requested `key` is not set in the config file.
+	 *
+	 * @param {String} key The key of the config entry.
+	 * @return {*} Value of config or error if not found.
+	 * @api public
+	 */
+	const config = key => {
+		if ( key in data ) {
+			return data[ key ];
+		}
+		throw new Error( 'config key `' + key + '` does not exist' );
```

Comment:
We should mirror the behavior changed in #11686 to prevent accidentally crashing large parts of the Calypso runtime when individual modules crash on import.